server/src/main/java/io/druid/curator/CuratorModule.java

```diff
@@ -91,6 +101,54 @@ public void stop()
     return framework;
   }
 
+  private CuratorFrameworkFactory.Builder newCuratorBuilder(CuratorConfig config, ExhibitorConfig exConfig)
+  {
+    CuratorFrameworkFactory.Builder builder = CuratorFrameworkFactory.builder();
+    Optional<EnsembleProvider> provider = newEnsembleProvider(exConfig, config.getZkHosts());
+    if (provider.isPresent()) {
+      return builder.ensembleProvider(provider.get());
+    }
+    return builder.connectString(config.getZkHosts());
+  }
+
+  private Optional<EnsembleProvider> newEnsembleProvider(ExhibitorConfig config, String backupZKHosts)
+  {
+    List<String> exhibitorHosts = config.getHosts();
+    if (exhibitorHosts.isEmpty()) {
+      return Optional.absent();
+    }
+    ExhibitorEnsembleProvider provider = new ExhibitorEnsembleProvider(
+        new Exhibitors(
+            exhibitorHosts,
+            config.getRestPort(),
+            newBackupProvider(backupZKHosts)
+        ),
+        new DefaultExhibitorRestClient(config.getUseSsl()),
+        config.getRestUriPath(),
+        config.getPollingMs(),
+        new BoundedExponentialBackoffRetry(1000, 45000, 10)
+    );
+    try {
+      provider.pollForInitialEnsemble();
+      return Optional.<EnsembleProvider>of(provider);
+    }
+    catch (Exception err) {
```

Comment:
```ExhibitorEnsembleProvider#pollForInitialEnsemble()``` should be called before calling ```CuratorFramework#start()```;
otherwise, the curator client will fall back to ```druid.zk.service.host``` at initial connect time.
Because the curator client immediately after start does not have a list of ZooKeeper servers.
There is a certain time lag before curator client gets a list of ZooKeeper servers.
https://github.com/apache/curator/blob/apache-curator-2.11.0/curator-client/src/main/java/org/apache/curator/ensemble/exhibitor/ExhibitorEnsembleProvider.java#L128

I will move the place to write ```ExhibitorEnsembleProvider#pollForInitialEnsemble()``` into ```Lifecycle.Handler#start()```.